### PR TITLE
Cyborg charger tweaks

### PR DESCRIPTION
**Pull request by MadmanMartian**
Something me and jkn have been discussing

Recharge stations now accept more than just cyborgs

Recharge stations now actually consume power from the grid

Recharge stations charge cells in their storage

:cl:
 * rscadd: Cyborg chargers now accept anyone that has a cell to them, not just silicons.
 * tweak: The cyborg charger now draws energy from the grid properly.
 * balance: The cyborg charger no longer fully charges a cell after upgrading.

### CODE REVIEW

File: code/game/machinery/rechargestation.dm [hunk 1]
```diff
@@ -257,18 +253,15 @@
...
-			if(R.module && R.module.modules)
-				var/list/um = R.contents|R.module.modules
-				// ^ makes single list of active (R.contents) and inactive modules (R.module.modules)
-				for(var/obj/item/I in um)
-					I.restock()
-				if(R)
-					if(R.module)
-						R.module.respawn_consumable(R)
-						R.module.fix_modules()
+	if(istype(occupant, /mob/living/silicon/robot))
```
Comment (jknpj) [inline code/game/machinery/rechargestation.dm]: isrobot()

**Comment by jknpj** [comment]
Maybe it's time to change its name to just "recharging station"?

**Comment by tuypo1** [comment]
Not so sure about accepting non cyborgs but the other changes sound good

**Comment by Kharkov72** [comment]
what else even CAN use these?

**Comment by MadmanMartian** [comment]
It's laying out the ground work for something else

**Comment by FudgePucker** [comment]
Does this make recharging slower or faster? Is it based on the amount in the grid?

**Comment by MadmanMartian** [comment]
It's the same speed, it just actually draws power from the grid now.

**Comment by FudgePucker** [comment]
Does it require a knot under the machine then?

**Comment by MadmanMartian** [comment]
No, use_power uses the APC for that area.

**Comment by MadmanMartian** [comment]
ty, keep forgetting to come back to this

**Pull request merged**